plugins/git/plugin.vala

```diff
@@ -46,17 +46,44 @@ public class Marlin.Plugins.Git : Marlin.Plugins.Base {
         Ggit.Repository git_repo = repo_map.lookup (file.directory);
 
         if (git_repo != null) {
+            string file_path = file.location.get_path ();
             /* Ignore other files specified by .gitignore */
             try {
-                if (git_repo.path_is_ignored (file.location.get_path ())) {
+                if (git_repo.path_is_ignored (file_path)) {
                     return;
                 }
             } catch (GLib.Error e) {
                 return; /* If this fails then unlikely to be able to get git_status */
             }
 
-            var git_status = update_git_status (file.location, git_repo, file.is_directory);
-            switch (git_status) {
+            Ggit.StatusFlags new_flag = Ggit.StatusFlags.CURRENT;
+            try {
+                if (file.is_directory) {
+                    git_repo.file_status_foreach (new Ggit.StatusOptions (0, Ggit.StatusShow.WORKDIR_ONLY, null), (path, status_flags) => {
```

Comment:
This matches too many files resulting in modified emblems being applied to directories which contain no modified files.
On solution is to include a pathspec parameter `{pathspec}` instead of `null`, where `var pathspec = "*".concat (file.basename, "*");`.